code/controllers/subsystem/ships.dm

```diff
@@ -128,13 +128,16 @@ SUBSYSTEM_DEF(ship)
 
 /datum/controller/subsystem/ship/proc/attack_player(var/datum/starship/S, var/datum/component/weapon/W)
 	var/datum/ship_attack/attack_data = W.attack_data
-	
+
 	if(prob(player_evasion_chance))
 		broadcast_message("<span class=notice> Enemy ship ([S.name]) fired their [W.name] but missed!</span>",success_sound,S)
 	else
 		if(SSstarmap.ftl_shieldgen && SSstarmap.ftl_shieldgen.is_active())
-			SSstarmap.ftl_shieldgen.take_hit()
-			broadcast_message("<span class=warning>Enemy ship ([S.name]) fired their [W.name] and hit! Hit absorbed by shields.",error_sound,S)
+			if(W.attack_data.shield_damage)
+				SSstarmap.ftl_shieldgen.take_hit()
+				broadcast_message("<span class=warning>Enemy ship ([S.name]) fired their [W.name] and hit! Hit absorbed by shields.",error_sound,S)
+			else
+				broadcast_message("<span class=warning>Enemy ship ([S.name]) fired their [W.name] and hit! Hit deflected by shields.",error_sound,S)
```

Comment:
Probably should be `Enemy ship ([S.name]) fired their [W.name] but it was deflected by the shields.` to make it clear this is a good thing